### PR TITLE
Keep single quotes for string literals instead of escaping double quotes inside

### LINKED ISSUE #298: .bzl formatter should preserve single-quoted strings when it would avoid backslash-escaping
One change that the formatter currently makes that I don't really agree with:
It changes single-quoted strings to double-quoted strings even in cases where it means adding backslashes, thus hindering readability.

* Example:
  * Old: `key_value_pairs = no_curly_braces.split('", "')`
  * New: `key_value_pairs = no_curly_braces.split("\", \"")`

Compare with the guidance in https://google.github.io/styleguide/pyguide.html#Strings that:
> It is okay to use the other quote character on a string to avoid the need to \ escape within the string.

### PR DESCRIPTION
**Pull request by vladmos**
Fixes #298.

### CODE REVIEW

File: build/print.go [hunk 1]
```diff
@@ -395,13 +395,14 @@ func (p *printer) expr(v Expr, outerPrec int) {
...
-		// If the Token is a correct quoting of Value, use it.
-		// This preserves the specific escaping choices that
-		// BUILD authors have made, and it also works around
-		// b/7272572.
-		if strings.HasPrefix(v.Token, `"`) {
-			s, triple, err := unquote(v.Token)
-			if s == v.Value && triple == v.TripleQuote && err == nil {
+		// If the Token is a correct quoting of Value and has double quotes, use it.
+		// This preserves the specific escaping choices that BUILD authors have made,
+		// and it also works around b/7272572.
```
Comment (pmbethe09) [inline build/print.go]: replace b/ link with a github issue

File: build/testdata/002.build.golden [hunk 2]
```diff
@@ -1,5 +1,5 @@
 cc_test(
-    name = "b\"ar'\"",
+    name = 'b\"ar\'"',
```
Comment (laurentlb) [inline build/testdata/002.build.golden]: Why do we escape the first `"` and not the second?

Comment (vladmos) [reply build/testdata/002.build.golden]: The original string is also `'b\"ar\'"'`. Buildifier currently doesn't remove unnecessary escapes: `"'\'"` is not changed during formatting.

File: build/print.go (hunk 1 above)
Comment (vladmos) [reply build/print.go]: Maybe I should just remove it? This is an old bug that has been fixed already.

File: build/testdata/002.build.golden (hunk 2 above)
Comment (vladmos) [reply build/testdata/002.build.golden]: You mean, inside single-quoted strings?

**Comment by vladmos** [comment]
@pmbethe09 this change doesn't affect already formatted BUILD files because it just allows more flexibility: if a string is originally wrapped in `'` it may remain formatted as it is if there's a reason. But files that have already been formatted with previous version of buildifier don't have such strings.

File: build/testdata/002.build.golden (hunk 2 above)
Comment (vladmos) [reply build/testdata/002.build.golden]: In this example `\"` should be actually replaced with unescaped `"`, at least that's how the Bazel interpreter parses strings.

Comment (laurentlb) [reply build/testdata/002.build.golden]: Oh right. The backslash can be dropped here.

The bug I was mentioning is for other characters: `"a\w" == "a\\w"`.

File: build/print.go (hunk 1 above)
Comment (laurentlb) [reply build/print.go]: Yes, remove.

Comment (vladmos) [reply build/print.go]: Done.

**Pull request merged**